### PR TITLE
Refactor ElasticsearchExporter integration tests

### LINKED ISSUE #9321: Refactor the ElasticsearchExporter tests
**Description**

The main goals here were:

- Cover as much as possible via only unit tests. This gives us a faster feedback cycle when changing something - developers can focus on first fixing unit tests, which have smaller blast radius and faster execution, before even looking at integration tests.
- Narrow the scope of the integration tests as much as possible. This meant getting rid of the broker, workflow engine semantics, etc., and just testing the integration of the exporter instance with Elasticsearch.

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR refactors the ES exporter integration tests into two different classes: `ElasticsearchExporterIT`, and `FaultToleranceIT`. The latter is split into its own as it needs to manipulate the Elastic instance during the test, meaning it may not play nice with the other tests.

The bulk of the tests are in `ElasticsearchExporterIT`. These tests share a single static Elastic container, and use a different, unique index prefix (`config.index.prefix`) for test isolation. Most of the pre-existing tests were rolled into these ones. In the future, it's expected most tests will also go there as well.

## Related issues

closes #9321 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/FaultToleranceIT.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      // when
+      ElasticsearchExporterException connectException = null;
+      try {
+        exporter.export(record);
+      } catch (final ElasticsearchExporterException e) { // expected since ES is down
+        connectException = e;
+      }
+      assertThat(connectException)
+          .as("sanity check: should have failed to export since ES was down")
+          .isNotNull();
```
Comment (remcowesterhoud) [inline exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/FaultToleranceIT.java:74]: 🔧 Replace with `assertThatThrownBy`

**Review by remcowesterhoud — APPROVED** [review]
Looking good or the most part. I have some small suggestions. Please have a look and merge after 👍

**Comment by npepinpe** [comment]
bors merge

**Comment by npepinpe** [comment]
Seems to be a broken cache thing again.

**Comment by npepinpe** [comment]
bors retry

**Pull request merged**